examples/pytorch/gcn/gcn_ls.py

```diff
@@ -0,0 +1,281 @@
+'''
+Adapted from Ziyue's code.
+'''
+
+
+import argparse, time, math
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+import dgl
+from dgl import DGLGraph
+import dgl.function as fn
+from dgl.data import register_data_args, load_data
+import scipy as sp
+from dgl import utils
+
+
+def generate_rand_graph(n):
+    arr = (sp.sparse.random(n, n, density=0.1, format='coo') != 0).astype(np.int64)
+    return dgl.DGLGraph(arr, readonly=True)
+
+
+def sample_subgraph(g, seed_nodes, n_layers, layer_size):
+    induced_nodes = []
+    seeds = seed_nodes
+    parent_uv_edges_per_hop = []
+    for _ in range(n_layers):
+        for subg, aux in dgl.contrib.sampling.LayerSampler(g, 1000000, layer_size,
+                                                           neighbor_type='in',
+                                                           seed_nodes=np.array(seeds),
+                                                           return_seed_id=True):
+            #seed_ids = aux['seeds']
+            #print(g.in_edges(seeds, form='all'))
+            subg_src, subg_dst = subg.edges()
+            parent_nid = subg.parent_nid
+            src = parent_nid[subg_src]
+            dst = parent_nid[subg_dst]
+            parent_uv_edges_per_hop.append((src.asnumpy(), dst.asnumpy()))
+            #print((src, dst))
+            seeds = list(np.unique(src.asnumpy()))
+            induced_nodes.extend(list(parent_nid.asnumpy()))
+
+    subgraph = g.subgraph(list(np.unique(np.array(induced_nodes))))
+    #print(subgraph.parent_nid)
+    #print(parent_uv_edges_per_hop)
+    subg_uv_edges_per_hop = [(subgraph.map_to_subgraph_nid(src).asnumpy(),
+                              subgraph.map_to_subgraph_nid(dst).asnumpy())
+                             for src, dst in parent_uv_edges_per_hop]
+    #print(subg_uv_edges_per_hop)
+    return subgraph, subg_uv_edges_per_hop
```

Comment:
you don't need to call LayerSampler like this. Previously, neighbor sampling can't handle resampled nodes correctly. That's why @ZiyueHuang implement neighbor sampling like this. If your layer sampler can handle resampled nodes, you can directly use layer sampler.

BTW, this implementation generates only a single batch in an epoch. This isn't desired. We need to enable mini-batch training.